App.js

```diff
@@ -0,0 +1,587 @@
+import React from 'react';
+import {useState, useEffect} from 'react';
+import type {Node} from 'react';
+import {
+  SafeAreaView,
+  ScrollView,
+  StatusBar,
+  StyleSheet,
+  Text,
+  useColorScheme,
+  View,
+  Modal,
+  Image,
+  TouchableOpacity,
+  TextInput,
+  Pressable
+} from 'react-native';
+
+import CustModal from './components/custmodal'
+
+import {
+  Colors,
+  DebugInstructions,
+  Header,
+  LearnMoreLinks,
+  ReloadInstructions,
+} from 'react-native/Libraries/NewAppScreen';
+import SideMenu from 'react-native-side-menu';
+import GestureRecognizer from 'react-native-swipe-gestures';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import KeepAwake from 'react-native-keep-awake';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import moment from 'moment';
+
+const Section = ({children, title}): Node => {
+  const isDarkMode = useColorScheme() === 'dark';
+  return (
+    <View style={styles.sectionContainer}>
+      <Text
+        style={[
+          styles.sectionTitle,
+          {
+            color: isDarkMode ? Colors.white : Colors.black,
+          },
+        ]}>
+        {title}
+      </Text>
+      <Text
+        style={[
+          styles.sectionDescription,
+          {
+            color: isDarkMode ? Colors.light : Colors.dark,
+          },
+        ]}>
+        {children}
+      </Text>
+    </View>
+  );
+};
+
+const App: () => Node = () => {
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const backgroundStyle = {
+    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+  };
+
+  const nmodestyle = {
+    backgroundColor: nMode ? 'black' : 'white',
+  };
+
+  const getsData = async(key) => {
+    try {
+      return await AsyncStorage.getItem(key);
+    } catch (e) {
+      return null
+    }
+  };
+
+  const [isOpen, setisOpen] = useState(false);
+  const [resData, setresData] = useState([]);
+  const [ang, setAng] = useState(1);
+  const [nMode, setnMode] = useState(false);
+  const [canSwipe, setcanSwipe] = useState(false);
+  const [fSize, setfSize] = useState(21);
+  const [kAwake, setkAwake] = useState(false);
+  const [sahajPaath, setsahajPaath] = useState(false);
+  const [isLarivaar, setisLarivaar] = useState(false);
+  const [datePicker, setdatePicker] = useState(false);
+  const [dateSelected, setdateSelected] = useState(new Date(1598051730000));
+  const [AngPop, setAngPop] = useState(false);
+  const [AngModal, setAngModal] = useState(false);
+  const [dang, setdang] = useState(0);
+  const [dailyTotal, setdailyTotal] = useState(0);
+  const [dailyProgress, setdailyProgress] = useState(0);
+
+  const [today_date, settoday_date ] = useState(getsData("today_date"));
+  const [today_start, settoday_start ] = useState(getsData("today_start"));
+  const [today_read, settoday_read ] = useState(ang - today_start);
+  
+
+
+  var now = moment().format('MM-DD-YYYY');
```

Comment:
Should use blocked scoped variables (`let`, `const`). shouldn't need to use `var`.